File: type.py

import sys

def main():
    # Verifica se o número de argumentos é suficiente
    print("\033c\033[47;30m")
    if len(sys.argv) != 2:
        print("Use: python type.py <file name>")
        return

    filename = sys.argv[1]

    try:
        # Abre o arquivo passado como argumento
        with open(filename, 'r') as file:
            # Lê o arquivo linha por linha e imprime na consola
            for line in file:
                print(line, end='')
    except FileNotFoundError:
        print(f"Erro: file '{filename}' not find.")
    except Exception as e:
        print(f"error: {e}")

if __name__ == "__main__":
    main()

File: type.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    // Verifica se o número de argumentos é suficiente
    printf("\033c\033[47;30m");
    if (argc != 2) {
        printf("Uso: type <file name>\n");
        return 1;
    }

    // Abre o arquivo passado como argumento
    FILE *file = fopen(argv[1], "r");
    if (file == NULL) {
        printf("Erro on file: %s\n", argv[1]);
        return 1;
    }

    char line[256];

    // Lê o arquivo linha por linha e imprime na consola
    while (fgets(line, sizeof(line), file)) {
        printf("%s", line);
    }

    // Fecha o arquivo
    fclose(file);

    return 0;
}

